spire-flex/tests/agentHealthCheckIPv4BindAddress.yaml

```diff
@@ -0,0 +1,103 @@
+suite: test .agent.healthCheck.bindAddress 192.168.0.23
+set:
+  Release.Name: devcluster
+templates:
+  - agent-configmap.yaml
+  - agent-daemonset.yaml
+tests:
+  - it: "Should render the correct agent_config file"
+    template: agent-configmap.yaml
+    set:
+      agent.healthCheck.bindAddress: "192.168.0.23"
+    asserts:
+      - containsDocument:
+          apiVersion: "v1"
+          kind: "ConfigMap"
+          name: "RELEASE-NAME-agent-config"
+          namespace: "NAMESPACE"
+      - equal:
+          path: data.agent_config
+          value: |
+            agent {
+            }
+            
+            plugins {
+            }
+            
+            health_checks {
+              listener_enabled = true
+              bind_address = "192.168.0.23"
```

Comment:
Something to consider.... 
if net=host, then the ip will be for each node its running on. which, unless only on a single node cluster, the ip will not be known until after its started, so unconfigurable here.
if not net=host, ip will be assigned by the cluster and also unknown until the pod is running.

So, a useful general spire option, but not as useful when running in k8s. Or, if the option really is desired, it maybe should specify an interface instead and something at runtime used to set the bind_address based on the ip on the interface.

Reply:
I constrained the values to localhost, 127.0.0.1, and ::1.  As we are using a DaemonSet for the agents, each Agent gets the same config file.  If localhost doesn't work, nothing else will.

Reply:
what I'm saying is, I dont think it will in some cases. setting it to 0.0.0.0 is an option that would always work.

Reply:
Also, please let me resolve the conversation when its resolved rather then assuming your statement actually does resolve the issue.

Reply:
Thanks. I read this in the wrong order.  I'll put in a quick PR to make 0.0.0.0 an additional option.  I'll keep the localhost options, because it's another of the rare set that might work.  Clearly any non-guaranteed to be present IP address won't work.  

I admit that when I started out on this item, I was more worried about getting the framework / unit testing / values schema / etc. working than having a working deployment, so I picked the agent first under the belief it would be easier.  Then I picked an easy thing to configure in the agent, the config file, then an easy stanza in the config.

Now that some patterns exist, new items should be faster to implement if the patterns hold and are reusable.

I believe the liveness probes already test within the pod's port numbering scheme, so I think localhost might work too, but certainly 0.0.0.0 should be present.

I'll also have you close out the conversations you start.  Keep in mind that if I change the section associated with the conversation, it auto closes the conversation (default github is to assume it was resolved by the code change, I guess).  That said, I did manually close out a few where I thought the answers were enough, and that's my mistake.  I'm sorry to have been presumptuous in that matter, and it won't happen again.

Reply:
kubelet on the host with its ip reaches through the vxlan.calico bridge (or whatever your network plugin does) to the pods ip. bind something in the pod to localhost, and the kubelet cant get to it.

You can see this with this example:
```
apiVersion: v1
kind: Pod
metadata:
  name: test
spec:
  containers:
  - name: main
    image: pnnlmiscscripts/debug-toolbox:latest
    command: ["nc", "-k", "-l", "127.0.0.1", "6000"]
    ports:
    - name: default
      containerPort: 6000
    readinessProbe:
      tcpSocket:
        port: default
      initialDelaySeconds: 1
      periodSeconds: 10
```
It never goes ready. You update 127.0.0.1 to -> 0.0.0.0 and the pod goes ready.